component/config.go

```diff
@@ -160,20 +163,40 @@ func MustNewType(strType string) Type {
 
 // DataType is a special Type that represents the data types supported by the collector. We currently support
 // collecting metrics, traces and logs, this can expand in the future.
-type DataType = Type
-
-func mustNewDataType(strType string) DataType {
-	return MustNewType(strType)
-}
+type DataType string
 
 // Currently supported data types. Add new data types here when new types are supported in the future.
-var (
+const (
 	// DataTypeTraces is the data type tag for traces.
-	DataTypeTraces = mustNewDataType("traces")
+	DataTypeTraces DataType = "traces"
 
 	// DataTypeMetrics is the data type tag for metrics.
-	DataTypeMetrics = mustNewDataType("metrics")
+	DataTypeMetrics DataType = "metrics"
 
 	// DataTypeLogs is the data type tag for logs.
-	DataTypeLogs = mustNewDataType("logs")
+	DataTypeLogs DataType = "logs"
 )
+
+func (dt DataType) String() string {
+	return string(dt)
+}
+
+func (dt DataType) MarshalText() (text []byte, err error) {
+	return []byte(dt), nil
+}
+
+func newDataType(ty string) (DataType, error) {
+	if len(ty) == 0 {
+		return "", errors.New("id must not be empty")
+	}
+	switch ty {
+	case "metrics":
+		return DataTypeMetrics, nil
+	case "logs":
+		return DataTypeLogs, nil
+	case "traces":
+		return DataTypeTraces, nil
+	default:
+		return "", fmt.Errorf("invalid data type %s", ty)
+	}
+}
```

Comment:
This smells like an [`encoding.TextUnmarshaler`](https://pkg.go.dev/encoding#TextUnmarshaler) implementation